### ISSUE TITLE
Control Client-Organizations

### ISSUE THREAD

**Issue by cwiechmann** [Axway-API-Management-Plus/apimanager-swagger-promote#12]
Adding the ability to control which organizations should have access to the API. 

The API-Developer will control it like this with the API-Contract:
```json
{
   "name":"My great API",
   "path":"/api/v1/path/to/something",
   "state":"published",
   "version":"1.0.1",
   "organization":"API Development",
   "clientOrganizations":[
      "Consuming Org 1",
      "Consuming Org 2",
      "Consuming Org 3"
   ]
}
```
To give permission to all existing organizations:
```json
{
   "name":"My great API",
   "path":"/api/v1/path/to/something",
   "state":"published",
   "version":"1.0.1",
   "organization":"API Development",
   "clientOrganizations":[
      "ALL"
   ]
}
```
Defining organizations will follow the same principle as all other API-Properties The API-Developer defines the desired-state, which will be replicated into the API-Manager. 
In order to identify the __actual__ state, the tool needs to iterate over all defined Orgs and get a list of APIs per org using the API-Manager REST-API.
With that the tool will perform only the required changes and perhaps do nothing, if all defined orgs already have permission to the API.

If an organization defined in the API-Contract isn't configured in the API-Manager, the tool will return with a unique error-code and print out all available organizations.

If the API-State is "unpublished", all configured organizations will be skipped and information is logged.

The tool will provide an option to ignore all configured organizations. This might be useful for instance to avoid on the production environment, that developers can control API-Permissions themselves.

**Comment by cwiechmann** [comment]
Some questions to the team (@rchinthakuntla, @rathnapandi, @rekies:
- What should happen, if a defined Org doesn't exists?
   - Should it:
      - Just skip this org? (I do prefer this one with a dedicated return code)
      - Fail completely?
      - Should there be an option, that the Org will be created "On-Demand"?
- I plan to add an Option: "ALL" (instead of the org-name), to assign permissions to all orgs. Agree?

Any other thoughts on this?

Chris

**Comment by rathnapandi** [comment]
Default behavior should be fail completely.  If user uses force flag, ignore the organization which does not exist on target APIM.

**Comment by rchinthakuntla** [comment]
Chris, 

I got a better idea on how to handle invalid organizations. For sure, we wan to throw error. As part of error message, can we show what the valid organizations are (both development and consumer)? Remember the startInstance script we provide in API Gateway? When I try to run it with invalid group or instance name, it errors out but provides information with the valid options. We can follow the same model here. This way, the team owing this CLI tool can correct the script and rerun it (instead of signing into Web UI and see what the valid organizations are). 

Makes sense? 

Thanks
Ravee

**Comment by cwiechmann** [comment]
Yes, makes sense and I have updated the description how I plan to implement it.

**Comment by rekies** [comment]
I really like the idea of failing and getting back the options as a result. It's a real time saver and production boost for CLI.

René

**Comment by cwiechmann** [comment]
Merge support for Organizations control into develop branch. Will be part of the next release.

**Comment by cwiechmann** [comment]
Just released version 1.3 which supports management of Client-Organizations.
